[Java] Added splicing example for IntVector (#237)
apache/arrow-cookbook: java/source/create.rst

    [[1,2,3], [10,20,30], [100,200,300], [1000,2000,3000]]

Splicing
========

Splicing provides a way of copying a range of rows between two vectors of the same type.

Splicing IntVector
------------------

In this example, we copy a portion of the input IntVector to a new IntVector.

.. testcode::

>     import org.apache.arrow.memory.BufferAllocator;

Reviewer: Just one thing: everything here is indented by one space too many I think (it should line up). Same with the output below

Author: There is similar comment on other PR, shall fix this once the other is fixed as expected :)

Reviewer: The other PR is now merged

Author: Done, hopefully it shall be correct this time.

.. _`FieldVector`: https://arrow.apache.org/docs/java/reference/org/apache/arrow/vector/FieldVector.html
.. _`ValueVector`: https://arrow.apache.org/docs/java/vector.html

> .. _`dictionary-encoding`: https://arrow.apache.org/docs/format/Columnar.html#dictionary-encoded-layout

Reviewer 2: Only by readability could be possible to move these lines to end of file

Author: Sure thing, let me move them to end and update the PR. Thanks

Author: updated

Splicing
========

> Splicing provides a way of copying complete or partial data between same types of Vectors

Reviewer: ```suggestion
Splicing provides a way of copying a range of rows between two vectors of the same type.
```

Splicing IntVector
------------------

> In this example, we want to copy a portion of input Vector to a new Vector.

Reviewer: ```suggestion
In this example, we copy a portion of the input IntVector to a new IntVector.
```

.. testcode::

    import org.apache.arrow.memory.BufferAllocator;
    import org.apache.arrow.memory.RootAllocator;
    import org.apache.arrow.vector.IntVector;
    import org.apache.arrow.vector.util.TransferPair;

>     try (BufferAllocator allocator = new RootAllocator();
>              IntVector vector = new IntVector("intVector", allocator);) {
>             for (int i = 0; i < 10; i++) {
>                 vector.setSafe(i, i);
>             }
>             vector.setValueCount(10);
> 
>             TransferPair tp = vector.getTransferPair(allocator);
>             tp.splitAndTransfer(0, 5);
>             IntVector sliced = (IntVector) tp.getTo();
>             System.out.print(sliced);
>             sliced.clear();
>         }

Reviewer 2: ```suggestion
    try (BufferAllocator allocator = new RootAllocator();
         IntVector vector = new IntVector("intVector", allocator)) {
        for (int i = 0; i < 10; i++) {
            vector.setSafe(i, i);
        }
        vector.setValueCount(10);

        TransferPair tp = vector.getTransferPair(allocator);
        tp.splitAndTransfer(0, 5);
        try (IntVector sliced = (IntVector) tp.getTo()) {
            System.out.print(sliced);
        }
    }
```

Reviewer: Maybe show slicing from a non-zero index?

Author: @lidavidm more examples coming
